src/main/java/com/google/api/generator/engine/ast/StringObjectValue.java

```diff
@@ -0,0 +1,39 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.api.generator.engine.ast;
+
+import com.google.auto.value.AutoValue;
+
+@AutoValue
+public abstract class StringObjectValue implements ObjectValue {
+
+  @Override
+  public TypeNode type() {
+    return TypeNode.STRING;
+  }
+
+  public abstract String value();
+
```

Comment:
Since value() is the only thing callers will set, let's add a `withString` or `withValue` method.

Reply:
sure, done.

Reply:
For learning purpose, why do we need `withValue` to set? @miraleung

Reply:
I guess I know a little bit. So it allow users has short cut set method to set string value.
Use `StringObjectValue.withValue()` instead of `StringObjectValue.builder().setValue("\"test\"").build();`

Reply:
Yup, that's it. :)